include/sway/scene_descriptor.h

```diff
@@ -0,0 +1,17 @@
+#ifndef _SWAY_SCENE_DESCRIPTOR_H
+#define _SWAY_SCENE_DESCRIPTOR_H
+#include <wlr/types/wlr_scene.h>
+
+enum sway_scene_descriptor_type {
+};
+
+bool scene_descriptor_assign(struct wlr_scene_node *node,
```

Comment:
It's called assign because it can only be set once. If you try to assign the same scene descriptor to the same node, it will abort. (The abort happens in wlroots: wlr_addon is the thing that doesn't allow this)

Reply:
Got it. I've usually seen this called `_set_once` in that case to clearly communicate the semantics, but I don't feel very strongly about this.